platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/GuestClaimsHandlerExt.java

```diff
@@ -52,72 +71,46 @@
 
     private Map<String, String> availableClaimsMap;
 
-    private List<Map<String, String>> profileMapsRaw;
-
-    private Map<String, Object> profilesProcessed;
-
-    private Map<String, Object> profileBannersProcessed;
-
     private final PropertiesFileReader propertiesFileReader;
 
     // ** Holds names of claims that if modified would cause guest login failures **
     private final List<String> immutableClaims;
 
-    private final String profilesDirectory;
-
-    private final String profilesBannerDirectory;
-
     private final String availableClaimsFile;
 
     private String selectedClaimsProfileName;
 
+    private final String profilesFilePath;
+
+    private Map<String, Object> profiles;
+
     /**
      * Constructor.
      */
     public GuestClaimsHandlerExt(PropertiesFileReader propertiesFileReader,
-            List<String> immutableClaims, String profilesDirectory, String profilesBannerDirectory,
-            String availableClaimsFile) {
+            List<String> immutableClaims, String availableClaimsFile, String profilesFilePath) {
 
         this.propertiesFileReader = propertiesFileReader;
         this.immutableClaims = ImmutableList.copyOf(immutableClaims);
-        this.profilesDirectory = profilesDirectory;
-        this.profilesBannerDirectory = profilesBannerDirectory;
         this.availableClaimsFile = availableClaimsFile;
+        this.profilesFilePath = profilesFilePath;
 
-        selectedClaimsProfileName = null;
+        this.selectedClaimsProfileName = null;
+        this.profiles = null;
```

Comment:
Any reason to not initialize this to empty map instead of null to prevent possible NPE's?